File: dataset/human/python/file_415.py
import numpy as np
import functools
import math
import scipy
import fractions
import itertools

def solve():
    n = int(input())
    a = list(map(int, input().split()))
    hoge = 0
    ans = 0
    for i in range(n):
        if a[i] < 0:
            hoge += 1
    if 0 in a or hoge % 2 == 0:
        for i in range(n):
            ans += abs(a[i])
    else:
        huga = []
        for i in range(n):
            a[i] = abs(a[i])
            ans += abs(a[i])
            if a[i] > 0:
                huga.append(a[i])
        ans -= min(huga)*2
    print(ans)
    return 0

if __name__ == "__main__":
    solve()

File: dataset/human/java/file_415.java
import java.util.*;

public class Main {
	public static void main (String[] args) {
		Scanner sc = new Scanner(System.in);
		int t = sc.nextInt();
		int d = sc.nextInt();
		int ta = sc.nextInt();
		int tb = sc.nextInt();
		double da = sc.nextInt();
		double db = sc.nextInt();
		double min = Double.MAX_VALUE;
		for (int i = 0; i * da <= d; i++) {
		    for (int j = 0; i * da + j * db <= d; j++) {
		        if (i == 0 && j == 0) {
		            continue;
		        }
		        min = Math.min(min, Math.abs((ta * i * da + tb * j * db) / (i * da + j * db) - t));
		    }
		}
		System.out.println(min);
	}
}
